src/EnergyPlus/SetPointManager.hh

```diff
@@ -1009,6 +905,27 @@ namespace SetPointManager {
         void calculate(EnergyPlusData &state);
     };
 
+    struct DefineSysNodeResetSetPointManager : SPBase // Derived type for System Node Reset Setpoint Manager Data
+    {
+        // Members
+        Real64 SpAtLowRef;  // Setpoint at Low Reference Temperature or Humidity Ratio (i.e., Maximum Temperature/Humidity Ratio Setpoint)
+        Real64 SpAtHighRef; // Setpoint at High Reference Temperature or Humidity Ratio (i.e., Maximum Temperature/Humidity Ratio Setpoint)
+        Real64 LowRef;      // Low Reference Temperature or Humidity Ratio
+        Real64 HighRef;     // High Reference Temperature or Humidity Ratio
+        int RefNodeNum;     // Reference Node Number
+        int NumCtrlNodes;
+        std::string CtrlNodeListName;
+        Array1D_int CtrlNodes;
+        Real64 SetPt; // current setpoint value
+
+        // Default Constructor
+        DefineSysNodeResetSetPointManager() : SpAtLowRef(0.0), SpAtHighRef(0.0), LowRef(0.0), HighRef(0.0), RefNodeNum(0), NumCtrlNodes(0), SetPt(0.0)
+        {
+        }
+
+        void calculate(EnergyPlusData &state);
+    };
+
```

Comment:
Single new `struct` for both SPMs.

Reply:
By the way, this module is absolutely begging for an object-oriented refactor. I am guessing you can reduce the amount of code by 50-75%. For calibration purposes, I estimate that the average EnergyPlus file is probably in the 25-33% range in terms of potential for code reduction from object-oriented refactoring.